### PR TITLE
Registry api code migration

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Migrates code previously resident in [registry-api's support subtree](https://github.com/NASA-PDS/registry-api/tree/main/support/provenance)

## ⚙️ Test Data and/or Report
No tests (yet).

Dockerfile has been modified for faster/easier dev experience.

## ♻️ Related Issues
fixes #1 
fixes #2

### CODE REVIEW

File: src/pds/registrysweepers/provenance.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    major_version, minor_version = lidvid.split("::")[1].split(".")
+    return (int(major_version), int(minor_version))
+
+
+def configure_logging(filepath: Union[str, None], log_level: int):
+    logging.root.handlers = []
+    handlers: List[logging.StreamHandler[TextIO]] = [logging.StreamHandler()]
+
+    if filepath:
+        handlers.append(logging.FileHandler(filepath))
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/provenance.py:81]: @sonatype-lift ignore

File: src/pds/registrysweepers/provenance.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+
+def _vid_as_tuple_of_int(lidvid: str):
+    major_version, minor_version = lidvid.split("::")[1].split(".")
+    return (int(major_version), int(minor_version))
+
+
+def configure_logging(filepath: Union[str, None], log_level: int):
+    logging.root.handlers = []
+    handlers: List[logging.StreamHandler[TextIO]] = [logging.StreamHandler()]
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/provenance.py]: @sonatype-lift ignore

File: src/pds/registrysweepers/provenance.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    major_version, minor_version = lidvid.split("::")[1].split(".")
+    return (int(major_version), int(minor_version))
+
+
+def configure_logging(filepath: Union[str, None], log_level: int):
+    logging.root.handlers = []
+    handlers: List[logging.StreamHandler] = [logging.StreamHandler()]
+
+    if filepath:
+        handlers.append(logging.FileHandler(filepath))
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/provenance.py:81]: @sonatype-lift ignore

File: docker/sweepers_driver.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+#   executable.
+#
+#
+
+import functools
+import json
+import logging
+import os
+
+from pds.registrysweepers import provenance
```
[the remark this answers is not in the document]
Comment (nutjob4life) [reply docker/sweepers_driver.py:62]: @alexdunnjpl regarding this, I don't know why Sonatype can't figure this out. It works for me:
```
$ .venv/bin/pip install  --quiet --editable '.[dev]'
$ .venv/bin/python -c 'from pds.registrysweepers import provenance'
$ echo $?
0
```

[the remark this answers is not in the document]
Comment (nutjob4life) [reply docker/sweepers_driver.py:62]: @sonatype-lift ignore

**Review by nutjob4life — APPROVED** [review]
Discussed extensively over slack as well!

**Pull request merged**